docs/ref/models/querysets.txt

```diff
@@ -1149,12 +1160,14 @@ that is that ``select_related('foo', 'bar')`` is equivalent to
 
 .. method:: prefetch_related(*lookups)
 
-Returns a ``QuerySet`` that will automatically retrieve, in a single batch,
-related objects for each of the specified lookups.
+Returns a ``QuerySet`` that will automatically retrieve the given lookups, each
+in one extra batch query. Prefetching is a way to optimize database access
+when you know you'll be accessing related objects later, so you can avoid
+triggering the on-demand loading behavior of the model instances’
+:doc:`fetch mode </topics/db/fetch-modes>`.
 
-This has a similar purpose to ``select_related``, in that both are designed to
-stop the deluge of database queries that is caused by accessing related objects,
-but the strategy is quite different.
+This method has a similar purpose to :meth:`select_related`, in that both are
+designed to eagerly fetch related objects. However, they work in different ways.
```

Comment:
I would expect chaining `.prefetch_related(None)` to clear a FETCH_PEERS fetch mode, but it doesn't -- it continues to fetch peers. What is your view on what should happen?

Reply:
currently the two API's are manipulating independent state, it'd feel weird to clear it like this instead of using fetch_mode
of course your other suggestion does couple them in another way

Reply:
Makes sense. (To my ear the docs advertised this as the successor to `prefetch_related()`, but looking again the docs clarify that there is a separate use case for `prefetch_related()`, e.g. in managers, so I'm convinced these are really two APIs after all.)

Reply:
Yeah, they are intended as two separate APIs. I tried to clarify that fetch modes apply to “on-demand fetches”, the existing behaviour where accessing an attribute fetches data. I think it’s clear enough, but if you have any suggested docs tweaks, I’m happy to see them.